server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -1468,7 +1468,7 @@ def ifthenelse(
                 case_=Case(),
             ).as_(step.new_column),
         )
-        return StepContext(query, columns + [step.new_column])
+        return StepContext(query, list(set(columns + [step.new_column])))
```

Comment:
LGTM but could we avoid the `set` usage to preserve column ordering ? Something like:
```suggestion
        return StepContext(query, columns + [step.new_column] if step.new_column not in columns else columns)
```